provider/mesh/peer.go

```diff
@@ -0,0 +1,273 @@
+package mesh
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/alertmanager/provider"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/log"
+	"github.com/prometheus/common/model"
+	"github.com/satori/go.uuid"
+	"github.com/weaveworks/mesh"
+)
+
+// NotificationInfos provides gossiped information about which
+// receivers have been notified about which alerts.
+type NotificationInfos struct {
+	st     *notificationState
+	send   mesh.Gossip
+	logger log.Logger
+}
+
+func NewNotificationInfos(logger log.Logger) *NotificationInfos {
```

Comment:
Meh, think we discussed these things in the past. Golint has way less
confidence than govet and commenting by saying "NewSilencer returns a new
silencer" seems slightly useless.

On Tue, Jun 21, 2016, 5:50 PM Björn Rabenstein <notifications@github.com>
wrote:

> In provider/mesh/peer.go
> <https://github.com/prometheus/alertmanager/pull/365#discussion_r67896005>
> :
>
> > +	"github.com/prometheus/alertmanager/types"
> > +	"github.com/prometheus/common/log"
> > +	"github.com/prometheus/common/model"
> > +	"github.com/satori/go.uuid"
> > +	"github.com/weaveworks/mesh
>
> "
> > +)
> > +
> > +// NotificationInfos provides gossiped information about which
> > +// receivers have been notified about which alerts.
> > +type NotificationInfos struct {
> > +	st     *notificationState
> > +	send   mesh.Gossip
> > +	logger log.Logger
> > +}
> > +
> > +func NewNotificationInfos(logger log.Logger) *NotificationInfos {
>
> doc comment to satisfy golint.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/prometheus/alertmanager/pull/365/files/656738451e34e0fcd9d8d2218133d2656718ee83#r67896005>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEuA8vapcygpZuxTVT089Y3uIxdT4Jc5ks5qOAhAgaJpZM4ItRTv>
> .
>